File: app/src/main/java/cjx/com/diary/api/HttpInterface.java
package cjx.com.diary.api;


import cjx.com.diary.mode.BaiDuImageBean;
import cjx.com.diary.mode.ImagesResult;
import bear.com.data.repository.db.model.NewsModel;
import cjx.com.diary.mode.test.RoomResult;
import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * @author: bear
 *
 * @Description: API接口定义
 *
 * @date: 2017/5/10
 */

public interface HttpInterface {
    @FormUrlEncoded
    @POST("/ModuleDefaultCompany/RentManage/SearchRentNo/")
    Observable<RoomResult> search(@Field("CertNo") String idNo);

    @FormUrlEncoded
    @POST(value = "http://www.tngou.net/tnfs/api/list")
    Observable<ImagesResult>getImages(@Field("page")int index,@Field("rows")int rows);

    @GET(value = "http://192.168.40.69:1377")
    Observable<NewsModel>getQiuBai();

    @GET(value = "http://image.baidu.com/channel/listjson?rn=10&ie=utf8")
    Observable<BaiDuImageBean>getBaiDuImage(@Query("pn") int index, @Query("tag1") String tag1,
                                            @Query("tag2") String tag2);
}

File: app/src/main/java/cjx/com/diary/api/HttpInterface.kt
package cjx.com.diary.api


import bear.com.data.repository.db.model.NewsModel
import cjx.com.diary.mode.BaiDuImageBean
import cjx.com.diary.mode.ImagesResult
import cjx.com.diary.mode.test.RoomResult
import io.reactivex.Observable
import retrofit2.http.*

/**
 * @author: bear
 *
 * @Description: API接口定义
 *
 * @date: 2017/5/10
 */

interface HttpInterface {

    @get:GET(value = "http://192.168.40.69:1377")
    val qiuBai: Observable<NewsModel>

    @FormUrlEncoded
    @POST("/ModuleDefaultCompany/RentManage/SearchRentNo/")
    fun search(@Field("CertNo") idNo: String): Observable<RoomResult>

    @FormUrlEncoded
    @POST(value = "http://www.tngou.net/tnfs/api/list")
    fun getImages(@Field("page") index: Int, @Field("rows") rows: Int): Observable<ImagesResult>

    @GET(value = "http://image.baidu.com/channel/listjson?rn=10&ie=utf8")
    fun getBaiDuImage(@Query("pn") index: Int, @Query("tag1") tag1: String,
                      @Query("tag2") tag2: String): Observable<BaiDuImageBean>
}
